app/controllers/chat_controller.rb

```diff
@@ -441,4 +465,11 @@ def create_action_whisper(topic, action)
         custom_fields: { "action_code_who" => current_user.username }
       )
   end
+
+  def find_channel_and_check_access(channel_id)
```

Comment:
Wonder if we shouldnt have some light service object for this Dont feel super good about this function.

Reply:
Yes that's fine, I only moved this into a function because it will be the second time I use it in this controller. Maybe we could have a class called `ChannelLookup`? Then we could have `ChannelLookup.find(id)` and it does the same thing as this function.

Not sure on the naming, any thoughts on what it might look like?

Reply:
I know we have ChannelFetcher already, didn't look much what it's doing at the moment, ChannelLookup sounds good to me